src/Fable.Core/Fable.Core.JS.fs

```diff
@@ -437,48 +437,180 @@ module JS =
         abstract warn: ?message: obj * [<ParamArray>] optionalParams: obj[] -> unit
         abstract table: ?data: obj -> unit
 
+    [<Obsolete("JS constructors are now in Fable.Core.JS.Constructors module to prevent conflicts with modules with same name")>]
     let [<Global>] Number: NumberConstructor = jsNative
+
+    [<Obsolete("JS constructors are now in Fable.Core.JS.Constructors module to prevent conflicts with modules with same name")>]
     let [<Global>] NaN: float = jsNative
```

Comment:
Does it make sense to move `NaN` and many others to `Constructors`?

Reply:
No, you're right. Actually I proposed the name `Constructors` to avoid moving NaN and other helpers like `encodeURI` because I think devs use those more often (at least I do) and have less risk of conflict. I was planning to change it before the release, thanks for the reminder!